vm/src/builtins/singletons.rs

```diff
@@ -33,8 +36,8 @@ impl<T: IntoPyObject> IntoPyObject for Option<T> {
 #[pyimpl]
 impl PyNone {
     #[pyslot]
-    fn tp_new(_: PyTypeRef, vm: &VirtualMachine) -> PyRef<Self> {
-        vm.ctx.none.clone()
+    fn tp_new(_: PyTypeRef, _args: FuncArgs, vm: &VirtualMachine) -> PyResult {
```

Comment:
Surprisingly, some types are too generous for `__new__` arguments, so they don't check them at all. But the other types check it. `None`, `Ellipsis` and `NotImplemented` are those types, so args type must be `()`